packages/protocol/test/ACE/ACEAdjustSupply.js

```diff
@@ -0,0 +1,354 @@
+                return note.create(publicKey, noteValues[i]);
+            });
+
+            // Creating a fixed note
+            const a = padLeft('1', 64);
+            const k = padLeft('0', 8);
+            const ephemeral = secp256k1.ec.keyFromPrivate(crypto.randomBytes(32));
+            const viewingKey = `0x${a}${k}${padLeft(ephemeral.getPublic(true, 'hex'), 66)}`;
+            zeroNote = note.fromViewKey(viewingKey);
+
+            const newTotalMinted = notes[0];
+            const oldTotalMinted = zeroNote;
+            const adjustedNotes = notes.slice(2, 4);
+
+            proofs[0] = proof.mint.encodeMintTransaction({
+                newTotalMinted,
+                oldTotalMinted,
+                adjustedNotes,
+                senderAddress: accounts[0],
+            });
+
+            const newTotalBurned = notes[0];
+            const oldTotalBurned = zeroNote;
+
+            proofs[1] = proof.burn.encodeBurnTransaction({
+                newTotalBurned,
+                oldTotalBurned,
+                adjustedNotes,
+                senderAddress: accounts[0],
+            });
+
+            const publicOwner = accounts[1];
+            const inputNoteOwners = aztecAccounts.slice(2, 4);
+
+            proofs[2] = proof.joinSplit.encodeJoinSplitTransaction({
+                inputNotes: adjustedNotes,
+                outputNotes: [],
+                senderAddress: accounts[0],
+                inputNoteOwners,
+                publicOwner,
+                kPublic,
+                validatorAddress: aztecJoinSplit.address,
+            });
+
+            erc20 = await ERC20Mintable.new();
+            const scalingFactor = new BN(1);
+            const canAdjustSupply = true;
+            const canConvert = true;
+
+            await erc20.mint(
+                accounts[0],
+                scalingFactor.mul(tokensTransferred),
+                { from: accounts[0], gas: 4700000 }
+            );
+
+            await erc20.approve(
+                ace.address,
+                scalingFactor.mul(tokensTransferred),
+                { from: accounts[0], gas: 4700000 }
+            );
+
+            await ace.createNoteRegistry(
+                erc20.address,
+                scalingFactor,
+                canAdjustSupply,
+                canConvert,
+                { from: accounts[0] }
+            );
+        });
+
+        it('successful call to the mint function', async () => {
+            const { receipt } = await ace.mint(MINT_PROOF, proofs[0].proofData, accounts[0]);
+            expect(receipt.status).to.equal(true);
+        });
+
+        it('successful burning of the minting notes', async () => {
+            await ace.mint(MINT_PROOF, proofs[0].proofData, accounts[0]);
+            const { receipt } = await ace.burn(BURN_PROOF, proofs[1].proofData, accounts[0]);
+            expect(receipt.status).to.equal(true);
+        });
+    });
+
+    describe('failure states', () => {
```

Comment:
We should add a test that creates 2 note registries linked to the same ERC20 token. In 1st registry, tokens (say, 'x' amount) converted into notes. In 2nd registry, a note is minted, and then converted into 'x' tokens. This must cause ACE to throw.

Reply:
Moved to testing checklist